### PR TITLE
catalog-react: move DefaultStarredEntitiesApi to catalog plugin + fixes

### PR DESCRIPTION
**Pull request by Rugvip**
We want to keep `catalog-react` lightweight, as it has broad usage and likely duplication. This moves `DefaultStarredEntitiesApi` over to the `catalog` plugin, but keeps the `StarredEntitiesApi` definition in `catalog-react`.

I also got rid of the unused `isStarred` method while at it, refactored the tests, and fixed a dangerous sharing of the mutated starred entities `Set`.

### CODE REVIEW

File: plugins/catalog-react/package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
   "devDependencies": {
     "@backstage/cli": "^0.14.0",
     "@backstage/core-app-api": "^0.5.3",
+    "@backstage/plugin-catalog": "^0.8.0",
```
Comment (freben) [inline plugins/catalog-react/package.json:65]: This feels like a wut. Is it really kosher, even if a dev dep?

**Comment by freben** [comment]
If I read things right, it seems that this pulls in the ScmAuthMux with uses private vars with a `#` prefix, which isn't handled by storybook so it all explodes. We'll probably have to address either the storybook build, or rewrite the auth mux to not use private vars.

**Review by jhaals — APPROVED** [review]

**Comment by emmaindal** [comment]
Hello @Rugvip, I see the example in `storybook` doesn't work anymore and my guess is the change in this PR, is this because the changes hasn't been released yet or something else?

**Comment by Rugvip** [comment]
@emmaindal doh! fixed in #9917

**Pull request merged**